model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -239,15 +286,15 @@ def _create_trial_objective(
         return objective
 
     def _get_objective_concurrency(self, trial_objectives: TrialObjectives) -> int:
-        concurrency = (
+        concurrency_formula = (
             2
             * int(trial_objectives["instance_group"])
             * int(trial_objectives["batch_sizes"])
         )
         concurrency = (
-            DEFAULT_RUN_CONFIG_MAX_CONCURRENCY
-            if concurrency > DEFAULT_RUN_CONFIG_MAX_CONCURRENCY
-            else concurrency
+            self._config.run_config_search_max_concurrency
```

Comment:
Bug fix. We should be respecting the max concurrency even when using the formula.